sema4ai/vscode-client/src/extension.ts

```diff
@@ -1172,3 +1170,98 @@ export async function dropDataSource(entry?: RobotEntry) {
+    if (!(await verifyDataExtensionIsInstalled())) {
+        return;
+    }
+    const dataServerStatus = await window.withProgress(
+        {
+            location: ProgressLocation.Notification,
+            title: "Getting data server status...",
+            cancellable: false,
+        },
+        async (progress: Progress<{ message?: string; increment?: number }>, token: CancellationToken) => {
+            const dataServerStatus = await commands.executeCommand(DATA_SERVER_STATUS_COMMAND_ID);
+            return dataServerStatus;
+        }
+    );
+
+    if (!dataServerStatus["success"]) {
+        window.showErrorMessage("Unable to get the data server status. Please start the data server and try again.");
+        return;
+    }
+
+    let selectedActionPackage: vscode.Uri | undefined;
+    if (!entry) {
+        selectedActionPackage = await findActionPackagePath({ includeSemaOrg: false });
+    } else {
+        selectedActionPackage = entry.uri;
+    }
+
+    if (!selectedActionPackage) {
+        return;
+    }
+
+    let listDataSourcesResult = await vscode.commands.executeCommand(SEMA4AI_LIST_ACTIONS_INTERNAL, {
+        "action_package": selectedActionPackage.toString(),
+        "collect_datasources": true,
+    });
+
+    if (!listDataSourcesResult["success"]) {
+        window.showErrorMessage(listDataSourcesResult["message"] || "Unknown error");
+    }
+
+    const dataSources: DatasourceInfo[] = listDataSourcesResult["result"].filter((ds) => {
+        return ds.kind === "datasource";
+    });
+
+    if (dataSources.length === 0) {
+        window.showInformationMessage("No Data Sources found.");
+        return;
+    }
+
+    const deleteExternalSources = await vscode.window.showInformationMessage(
+        `Do you want to remove external sources as well?`,
+        { modal: true },
+        "Yes",
+        "No"
+    );
+
+    if (deleteExternalSources === undefined) {
+        return;
+    }
+
+    await window.withProgress(
+        {
+            location: ProgressLocation.Notification,
+            title: "Dropping Data Sources...",
+            cancellable: true,
+        },
+        async (progress: Progress<{ message?: string; increment?: number }>, token: CancellationToken) => {
+            for (const item of dataSources.reverse()) {
+                if (token.isCancellationRequested) {
+                    OUTPUT_CHANNEL.appendLine("Operation cancelled.");
+                    return;
+                }
+
+                if (isExternalDatasource(item) && deleteExternalSources === "No") {
+                    continue;
+                }
+
+                const result = await langServer.sendRequest("dropDataSource", {
+                    "datasource": item,
+                    "data_server_info": dataServerStatus["data"],
+                });
+
+                if (!result["success"]) {
+                    window.showErrorMessage(result["message"] || "Unknown error");
```

Comment:
If you have an error it'll show this message and then `Data Sources dropped successfully` afterwards.